features/entities/memory.feature

```diff
@@ -0,0 +1,15 @@
+Feature: entities
+
+Scenario: boxes
+  Given a box B1 with { "color": "red" }
+  And a box B2 with { "color": "green" }
+  And box B2 also has { "color": "blue", "size": "big" }
+  Then the document for box B1 at color is "red"
+  And that document at id is 1
+  And the document for the box with { "color": "blue" } at size is "big"
+  And the box B3 was deleted
+
+Scenario: previous boxes should not exist
+  And the document for the box with { "color": "red" } does not exist in context
```

Comment:
This is a test of the `memory` entity adapter. As `mongodb` and `elasticsearch`, most of the scenarios do not have a `When` because it's just checking that saving a document to a specific store, can be queried with the defined `Then` steps from the adapter itself. It is actually testing the pickled-cucumber `memory entity module`. Of course this could be improved a lot, with all these tests we are just checking that `write` and `read` actions are synced up.